subprojects/core/src/main/java/org/gradle/api/internal/project/ProjectFactory.java

```diff
@@ -20,14 +20,17 @@
 import org.gradle.api.internal.GradleInternal;
 import org.gradle.api.internal.initialization.ClassLoaderScope;
 import org.gradle.groovy.scripts.ScriptSource;
-import org.gradle.groovy.scripts.UriScriptSource;
+import org.gradle.groovy.scripts.TextResourceScriptSource;
 import org.gradle.internal.reflect.Instantiator;
+import org.gradle.internal.resource.TextResource;
+import org.gradle.internal.resource.BasicTextResourceLoader;
 
 import java.io.File;
 
 public class ProjectFactory implements IProjectFactory {
     private final Instantiator instantiator;
     private final ProjectRegistry<ProjectInternal> projectRegistry;
+    private final BasicTextResourceLoader resourceLoader = new BasicTextResourceLoader();
```

Comment:
It _could_ but then we'd need 2 different `TextResourceLoader` types: one that can only load `File` types (and is available without of dependency management) and a fully fledged one with `URI` loading capabilities. It's probably a good idea, now that I write my response.